data/transaction/status.go

```diff
@@ -35,25 +35,29 @@ func (tx TxStatus) String() string {
 
 // StatusComputer computes a transaction status
 type StatusComputer struct {
-	MiniblockType            block.Type
-	IsMiniblockFinalized     bool
-	SourceShard              uint32
-	DestinationShard         uint32
-	Receiver                 []byte
-	TransactionData          []byte
-	SelfShard                uint32
-	HeaderNonce              uint64
-	HeaderHash               []byte
-	Uint64ByteSliceConverter typeConverters.Uint64ByteSliceConverter
-	Store                    dataRetriever.StorageService
+	SelfShard	uint32
+}
+
+func NewStatusComputer(selfShard uint32) *StatusComputer {
+	statusComputer := &StatusComputer{
+		SelfShard: selfShard}
+
+	return statusComputer
 }
 
 // ComputeStatusWhenInStorageKnowingMiniblock computes the transaction status for a historical transaction
-func (sc *StatusComputer) ComputeStatusWhenInStorageKnowingMiniblock() TxStatus {
-	if sc.isMiniblockInvalid() {
+func (sc *StatusComputer) ComputeStatusWhenInStorageKnowingMiniblock(
+	miniblockType block.Type,
+	tx *ApiTransactionResult) TxStatus {
+	isMiniblockFinalized := tx.NotarizedAtDestinationInMetaNonce > 0
+	destinationShard := tx.DestinationShard
```

Comment:
This variable could have been avoided. `tx.DestinationShard` expresses the intent.

Reply:
Done!